addons/knowledge/models/knowledge_article.py

```diff
@@ -1431,6 +1432,14 @@ def _send_invite_mail(self, partners):
                 email_layout_xmlid='mail.mail_notification_light'
             )
 
+    def _mail_track(self, tracked_fields, initial):
+        changes, tracking_value_ids = super(Article, self)._mail_track(tracked_fields, initial)
+        if {'parent_id', 'root_article_id'} & changes and not self.user_has_write_access:
+            partner_name = self.env.user.partner_id.display_name
+            message_body = _("%(partner)s had no access to this Article so we preserved it here.", partner=partner_name)
```

Comment:
This sentence is not really clear to me ... what do you mean by "we preserved it here" ? Anyway tracked fields are tracked. You could maybe say "Logging update from %(partner_name)s without write access on article %(article_name)s due to hierarchy tree update`` ?